src/stlink-lib/chipid.c

```diff
@@ -915,6 +916,22 @@ void dump_a_chip (FILE *fp, struct stlink_chipid_params *dev) {
     fprintf(fp, "flags %d\n\n", dev->flags);
 }
 
+static int chipid_params_eq(struct stlink_chipid_params *p1, struct stlink_chipid_params *p2)
```

Comment:
Please consider making the function parameters `const`:

```suggestion
static int chipid_params_eq(const struct stlink_chipid_params *p1, const struct stlink_chipid_params *p2)
```